src/sprite_weapon.cpp

```diff
@@ -80,13 +80,13 @@ void Sprite_Weapon::SetWeaponAnimation(int nweapon_animation_id) {
 }
 
 void Sprite_Weapon::StartAttack(bool secondary_weapon) {
-	cycle = 0;
+	auto* battler = GetBattler();
 
-	attacking = true;
+	SetZ(battler->GetBattleSprite()->GetZ() + ((secondary_weapon ^ battler->IsDirectionFlipped()) ? 1 : -1));
```

Comment:
New question. I missed this during the review:
Can you replace this xor with a if-else? I always have a really hard time to understand what is happening when XOR is used.
But please do not use multiple layers of "ternary ifs" to solve this :)

Reply:
Fixed (I hope the correct way).

Reply:
Looks okay